### PR TITLE
feat(search): Implementing SnubaOnlyQueryExecutor and SnubaGroupsSearchBackend

### PR DESCRIPTION
**Pull request by iProgramStuff**

### CODE REVIEW

File: src/sentry/search/snuba/executors.py
Comment (fpacifici) [inline src/sentry/search/snuba/executors.py]: This is not relevant anymore

File: src/sentry/search/snuba/executors.py
Comment (fpacifici) [inline src/sentry/search/snuba/executors.py]: Now calling this candidate_ids is out of contexts. Let's call it `group_ids`

File: src/sentry/search/snuba/executors.py
Comment (fpacifici) [inline src/sentry/search/snuba/executors.py]: Is this the list of differences between snuba_search in the context of PostgresSnubaQueryExecutor and SnubaOnlyQueryExecutor ?
self.QUERY_DATASET
self.TABLE_ALIAS
self.issue_only_fields
self. modify_converted_filter 
self. aggregation_defs 
?

If yes, then I will suggest to implement a template method for this. https://en.wikipedia.org/wiki/Template_method_pattern
This is very similar with what you already have here, just add protected getter methods for those fields instead of accessing the fields directly, so the parent class cannot change the state of the subclasses.
The goal of this approach is that the parent class keeps control of the execution of snuba_query (which makes sense because the vast majority of the code is common) but it defines (through abstract method) some specific extension points the subclasses can customize. By avoiding sharing the state directly you guarantee that, in order to understand the subclass you do not have to check whether the parent class makes change to the state, thus reducing the coupling.

File: src/sentry/search/snuba/backend.py
Comment (fpacifici) [inline src/sentry/search/snuba/backend.py]: Is this actually based on master now ?

File: src/sentry/search/snuba/executors.py
Comment (fpacifici) [inline src/sentry/search/snuba/executors.py]: Ok, now, you should be able to see that there is a lot of common logic between the two implementations of AbstractQueryExecutor. It seems to me these are:
- some parameters preparation (start/end/retention)
- chunked query (basically from line 386 to 414
- preparing the paginator to return at the end of this method
There are two ways to proceed to rationalize this:
1) you build a template method (see the comment about snuba_search). Which means, the parent class implement query and the subclasses implement the part that is not in common
2) as an alternative, you keep query in the subclasses and put the common parts into protected methods in the parent class.

your call on this. Either can work if meaningful methods are defined.

File: src/sentry/search/snuba/backend.py
Comment (fpacifici) [inline src/sentry/search/snuba/backend.py]: I think this is already in master. Could you please merge it.

**Review by fpacifici — CHANGES_REQUESTED** [review]

File: src/sentry/testutils/cases.py
Comment (markstory) [inline src/sentry/testutils/cases.py]: Does snuba not accept the `isoformat()` here?

File: src/sentry/utils/snuba.py
Comment (markstory) [inline src/sentry/utils/snuba.py]: This branch shouldn't run for transactions dataset, or discover. Instead this looks like it is only applicable for the groups dataset.

File: src/sentry/utils/snuba.py
Comment (markstory) [inline src/sentry/utils/snuba.py]: The groups dataset is the only one that needs prefixing, you should make this the special case in my opinion.

File: tests/snuba/search/test_backend.py
Comment (markstory) [inline tests/snuba/search/test_backend.py]: Why are there backticks in here?

**Review by markstory — COMMENTED** [review]
The current changes to `constrain_column_to_dataset` look like they will break discover2, the tests should be catching that regression as well.

File: src/sentry/search/snuba/backend.py
Comment (fpacifici) [reply src/sentry/search/snuba/backend.py]: Unresolving this because I don;t think the comment was addressed. 
This PR is not based on the current master. It seems it is missing major changes to snuba/backend.py that are included in PR https://github.com/getsentry/sentry/pull/15708/files that was already merged.

Please either rebase or merge master into this branch, till then, reviewing is not very useful.

**Review by fpacifici — CHANGES_REQUESTED** [review]
As I asked before, please rebase this on master or merge master in this branch. This does not seem to include the changes in https://github.com/getsentry/sentry/pull/15708/files

File: tests/snuba/search/test_backend.py
Comment (iProgramStuff) [reply tests/snuba/search/test_backend.py]: It came in from the groups dataset aggregations, which define this using `events.last_seen`, and without the ticks gives an error.

Now that they are more decoupled though, I can remove the ticks from this aggregation.

**Pull request closed without merging**
